### PR TITLE
Publish the server charm

### PR DESCRIPTION
**Pull request by mz2**
Publish the charm to charmhub

### CODE REVIEW

File: server/charm/metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  hardware information for a given machine, in particular helping you determine if a given hardware platform is Ubuntu Certified,
+  matches partially to a Certified system, or is not Certified.
+
+  For more information, see https://github.com/canonical/hardware-api
+  For schema, see https://canonical.github.io/hardware-api
+
+containers:
+  hardware-api:
+    resource: hardware-api-image
+
```
Comment (plars) [inline server/charm/metadata.yaml:17]: The build failure should be resolvable by removing the containers section from charmcraft.yaml

**Review by plars — CHANGES_REQUESTED** [review]

File: .github/workflows/publish_server.yml [hunk 2]
```diff
@@ -50,3 +50,19 @@ jobs:
...
+    needs: build-and-push-image
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Upload charm to charmhub
+        uses: canonical/charming-actions/upload-charm@2.4.0
+        with:
+          charm-path: "./server/charm"
```
[the remark this answers is not in the document]
Comment (mz2) [reply .github/workflows/publish_server.yml:65]: Indeed no practical effect, but let's be consistent. Fixed.

File: .github/workflows/publish_server.yml [hunk 3]
```diff
@@ -1,9 +1,9 @@
 on:
   push:
-    branches: ["main"]
+    branches: ["main", "charm-publishing"]
```
[the remark this answers is not in the document]
Comment (mz2) [reply .github/workflows/publish_server.yml:3]: Yep, will remove once there's one successful build.

**Review by plars — APPROVED** [review]
Ok, I at least got it to push to the charmstore now. The errors made no sense so I tried pushing it manually using the same credentials and it worked. After that, it was able to push the resource. So I'm wondering if maybe the charm has to be pushed at least once before the resource is pushed? but that won't work with the action alone because when the charm is pushed it also ties it to the resource for the release...
There are still errors, but I was at least able to confirm that the charm is in the store successfully now. So let's merge this and then we can keep iterating on it.

**Pull request merged**